docs/how-to-source-git.md

```diff
@@ -0,0 +1,180 @@
+# How to source-git?
+
+This is a practical introduction to source-git using a real example.
+
+If you are interested in the theory behind source-git, please read [the
+specification](/docs/source-git.md).
+
+
+## Let's create a source-git repo
+
+I choose systemd from Fedora 29 for this example.
+
+
+### What do we need?
+
+3 things:
+
+1. Systemd upstream repo.
+2. Fedora 29 dist-git repo of systemd.
+3. New local git repo.
+
+Let's set all of this up. We'll start with an empty git repository:
+```
+$ mkdir systemd-source-git
+$ cd systemd-source-git
+$ git init .
+Initialized empty Git repository in /home/tt/t/systemd-source-git/.git/
+```
+
+We'll add fedora and upstream remotes, and then we'll fetch them:
+```
+$ git remote add upstream https://github.com/systemd/systemd.git
+$ git remote add fedora ssh://ttomecek@pkgs.fedoraproject.org/rpms/systemd.git
```

Comment:
Isn't this a little bit confusing to have both upstream and downstream remotes in one git repo?

Reply:
yeah... https://github.com/TomasTomecek/pretty-git-prompt/issues/42

feel free to fix it, it was the easiest thing I was able to figure out

Reply:
Two different directories?

Reply:
for me it's simpler to have a single repo: yes, it's kinda confusing given the completely different histories, but I'm okay with that; anyone else has any opinion?

Reply:
Franto, I'm not ignoring your opinion, but I feel like it's not such a big deal, to use only a single repo; also I don't want to spend time on fixing it myself, so if it's a big deal for you, please provide me the suggestion and I can integrate it

Reply:
Definitely not such a big deal to spend more time on it for now. We can improve it later.